<?xml version="1.0" encoding="UTF-8"?>
<!--
  /**
   * This file is part of the Joule Plugin Code Checker suite.
   *
   * Joule Plugin Code Checker suite is free software: you can redistribute
   * it and/or modify it under the terms of the GNU General Public License as
   * published by the Free Software Foundation, either version 3 of the
   * License, or (at your option) any later version.
   *
   * Joule Plugin Code Checker suite is distributed in the hope that it will
   * be useful, but WITHOUT ANY WARRANTY; without even the implied
   * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
   * See the GNU General Public License for more details.
   *
   * You should have received a copy of the GNU General Public License
   * along with the Joule PLugin Code Checker suite.
   * If not, see <http://www.gnu.org/licenses/>
   *
   * @author    Corey Wallis <corey.wallis@blackboard.com>
   * @copyright 2015 Blackboard Inc.
   * @license   https://www.gnu.org/copyleft/gpl.html GPLv3
   * @link      https://github.com/techxplorer/joule-plugincheck
   */
-->
<documentation title="Warn about classes instantiation without namespaces">
  <standard>
    <![CDATA[
    Warn about attempts to instantiate classes that require namespaces.
    ]]>
  </standard>
  <code_comparison>
    <code title="Valid: Instantiation with namespaces">
        <![CDATA[
namespace pies;

use \pies\plates\Wedgwood;
use \pies\cutlery\Spoon as Spork;

class apple {

    protected $createdon;
    protected $plate;
    protexted $cutlery;

    function __construct(){
         $this->createdon = new DateTime();
         $this->plate = new Wedgwood();
         $this->cutlery = new Spork();
    }
}
]]>
    </code>
    <code title="Invalid: Instantiation without namespaces">
        <![CDATA[
namespace pies;

class apple {

    protected $createdon;
    protected $plate;
    protexted $cutlery;

    function __construct(){
         $this->createdon = new DateTime(); // <-- this should be new \DateTime();
         $this->plate = new Wedgwood();
         $this->cutlery = new Spork();
    }
}
]]>
    </code>
  </code_comparison>
</documentation>
